From 8da9a2d4346afef45cceedad66bf1c43b08f03af Mon Sep 17 00:00:00 2001
From: Tobias Heider <tobias.heider@canonical.com>
Date: Sat, 26 Oct 2024 14:09:55 +0200
Subject: [PATCH 35/56] UBUNTU: SAUCE: HACK: workaround for T14s OLED backlight

This makes the T14s OLED variant work with the LCD device tree.
Drop once we have a proper migration path to the correct dtb.

Original from flto at:
https://github.com/flto/linux/commit/aeca3a4bf5ac6598b30b8f602ddef7be302b3ec0

Signed-off-by: Tobias Heider <tobias.heider@canonical.com>
---
 drivers/gpu/drm/display/drm_dp_helper.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/display/drm_dp_helper.c b/drivers/gpu/drm/display/drm_dp_helper.c
index afe36cb8d85f..9fc152c16507 100644
--- a/drivers/gpu/drm/display/drm_dp_helper.c
+++ b/drivers/gpu/drm/display/drm_dp_helper.c
@@ -4286,6 +4286,14 @@ drm_edp_backlight_init(struct drm_dp_aux *aux, struct drm_edp_backlight_info *bl
 	if (ret < 0)
 		return ret;
 
+	if (bl->max == 0) {
+		/*
+		 * XXX: The T14s oled reports a broken max brightness of 0.
+		 * Actual maximum might be higher than 2047
+		 */
+		bl->max = 2047;
+	}
+
 	ret = drm_edp_backlight_probe_state(aux, bl, current_mode);
 	if (ret < 0)
 		return ret;
-- 
2.48.1

